### PR TITLE
Add support for non-root MySQL databases

### PR DESCRIPTION
**Pull request by Grendel7**
Fix for issue #194

### CODE REVIEW

File: scripts/bootstrap.py [hunk 1]
```diff
@@ -122,12 +122,16 @@ def init_seafile_server():
...
+        'MYSQL_USER': get_conf('DB_USER', 'seafile'),
+        'MYSQL_USER_PASSWD': get_conf('DB_USER_PASSWD', str(uuid.uuid4())),
+        'MYSQL_USER_HOST': get_conf('DB_USER_HOST', '%.%.%.%'),
+        'MYSQL_HOST': get_conf('DB_HOST', '127.0.0.1'),
+        'CCNET_DB': get_conf('CCNET_DB', 'ccnet_db'),
+        'SEAFILE_DB': get_conf('SEAFILE_DB', 'seafile_db'),
+        'SEAHUB_DB': get_conf('SEAHUB_DB', 'seahub_db'),
         # Default MariaDB root user has empty password and can only connect from localhost.
         'MYSQL_ROOT_PASSWD': get_conf('DB_ROOT_PASSWD', ''),
+        'USE_EXISTING_DB': get_conf('USE_EXISTING_DB', '0'),
```
Comment (GRBurst) [inline scripts/bootstrap.py]: This should make similar checks as other boolean flags, e.g. comparing to `'true', '1', 'yes'`.
So this becomes like
```'USE_EXISTING_DB': get_conf('USE_EXISTING_DB', '').lower() in ('true', '1', 'yes')```

**Comment by Neamar** [comment]
Hey there! So if I understand correctly:

* this PR is working and allows to use a non-root DB user
* it's not merged "for various reasons" (comment on #194), but is actually usable and working?

If so, I'm probably going to try setting it up :)

**Comment by mlazzarotto** [comment]
Bumping the thread, it's 2023 guys...

**Comment by freeplant** [comment]
The latest version already support using external database: https://manual.seafile.com/docker/deploy_seafile_with_docker/#use-an-existing-mysql-server

And DB_ROOT_PASSWD is needed during installation. Later, after Seafile is installed, the user seafile will be used to connect to the mysql-server (in conf/seafile.conf). You can remove the DB_ROOT_PASSWD.

**Pull request closed without merging**